bears/c_languages/CPPLintBear.py

```diff
@@ -1,32 +1,29 @@
-import re
-
-from coalib.bearlib.abstractions.Lint import Lint
-from coalib.bears.LocalBear import LocalBear
+from coalib.bearlib.abstractions.Linter import linter
 from coalib.settings.Setting import typed_list
 
 
-class CPPLintBear(LocalBear, Lint):
-    executable = 'cpplint'
-    output_regex = re.compile(
-        r'(?P<filename>.+\..+):(?P<line>\d+):\s(?P<message>.+)')
-    use_stderr = True
-
-    def run(self,
-            filename,
-            file,
-            max_line_length: int=80,
-            cpplint_ignore: typed_list(str)=[],
-            cpplint_include: typed_list(str)=[]):
-        '''
-        Checks the code with `cpplint` on each file separately.
+@linter(executable='cpplint',
+        use_stdout=False,
+        use_stderr=True,
+        output_format='regex',
+        output_regex=r'.+:(?P<line>\d+): (?P<message>.+)')
+class CPPLintBear:
+    """
+    Checks the code with ``cpplint`` on each file separately.
+    """
 
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         max_line_length: int=80,
+                         cpplint_ignore: typed_list(str)=[],
+                         cpplint_include: typed_list(str)=[]):
+        """
         :param max_line_length: Maximum number of characters for a line.
```

Comment:
yes the new linter allows you to put your parameters and documentation to arbitrary methods and merges it accordingly